_query-dsl/specialized/percolate.md

```diff
@@ -0,0 +1,620 @@
+
+## How percolation works
+
+1. Saved queries are stored in a special [`percolator` field type]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/percolator/).
+2. Documents are compared against all saved queries.
+3. Matching queries are returned with their `_id`.
+4. If highlighting is enabled, matched text snippets are also returned.
+5. If multiple documents are sent, `_percolator_document_slot` displays which document matched.
+
+## Create an index to store saved queries
+
+First, create an index with `mappings` configured with [`percolator` field type]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/percolator/) to store the saved queries:
+
+```json
+PUT /my_percolator_index
+{
+  "mappings": {
+    "properties": {
+      "query": {
+        "type": "percolator"
+      },
+      "title": {
+        "type": "text"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Add a query matching "apple" in `title` field:
+
+```json
+POST /my_percolator_index/_doc/1
+{
+  "query": {
+    "match": {
+      "title": "apple"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Add a query matching "banana" in `title` field:
+
+```json
+POST /my_percolator_index/_doc/2
+{
+  "query": {
+    "match": {
+      "title": "banana"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Percolate an inline document
+
+You can test an inline document against saved queries:
+
+```json
+POST /my_percolator_index/_search
+{
+  "query": {
+    "percolate": {
+      "field": "query",
+      "document": {
+        "title": "Fresh Apple Harvest"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+This will match the saved query looking for "apple":
+
+```json
+{
+  ...
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 0.13076457,
+    "hits": [
+      {
+        "_index": "my_percolator_index",
+        "_id": "1",
+        "_score": 0.13076457,
+        "_source": {
+          "query": {
+            "match": {
+              "title": "apple"
+            }
+          }
+        },
+        "fields": {
+          "_percolator_document_slot": [
+            0
+          ]
+        }
+      }
+    ]
+  }
+}
+```
+
+## Percolate an existing indexed document
+
+Create a separate index for your documents:
+
+```json
+PUT /products
+{
+  "mappings": {
+    "properties": {
+      "title": {
+        "type": "text"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Add a document:
+
+```json
+POST /products/_doc/1
+{
+  "title": "Banana Smoothie Special"
+}
+```
+{% include copy-curl.html %}
+
+Check if the stored queries match the indexed document:
+
+```json
+POST /my_percolator_index/_search
+{
+  "query": {
+    "percolate": {
+      "field": "query",
+      "index": "products",
+      "id": "1"
```

Comment:
Are both of these parameters always  used together ?

Reply:
I've added a note that both are required